modules/apigee_edge_teams/src/Plugin/Block/TeamContextSwitcherBlock.php

```diff
@@ -0,0 +1,273 @@
+    foreach ($entities as $entity) {
+      // If we are on an additional link route, then use the canonical url.
+      // Otherwise the destination url will be NULL and no link is rendered.
+      $url = is_string($current_context) ? $entity->toUrl() : $this->teamContextManager->getDestinationUrlForEntity($entity);
+
+      // Check for access and add link.
+      if ($url && $url->access($this->account)) {
+        $links[] = [
+          'title' => $entity->label(),
+          'url' => $url,
+          'is_current' => ($current_context instanceof EntityInterface && $current_context->getEntityTypeId() === $entity->getEntityTypeId() && $current_context->id() === $entity->id()),
+        ];
+      }
+    }
+
+    // Add additional links.
+    foreach ($this->getAdditionalLinks() as $route_name => $title) {
+      if (($url = Url::fromRoute($route_name)) && ($url->access($this->account))) {
+        $links[] = [
+          'title' => $title,
+          'url' => $url,
+          'is_current' => (is_string($current_context) && $current_context === $route_name),
+        ];
+      }
+    }
+
+    // Move the current link to the top and render it as a non link.
+    foreach ($links as $index => $link) {
+      if (isset($link['is_current']) && $link['is_current']) {
+        unset($links[$index]);
+        $link['url'] = Url::fromRoute('<nolink>');
+        array_unshift($links, $link);
+        break;
+      }
+      unset($link['is_current']);
+    }
+
+    return count($links) ? [
+      '#type' => 'dropbutton',
+      '#links' => $links,
+      '#attributes' => [
+        'class' => [
+          'team-switcher',
+        ],
+      ],
+      '#attached' => [
+        'library' => [
+          'apigee_edge_teams/switcher',
+        ],
+      ],
+    ] : [];
+  }
+
+  /**
+   * Returns an array of additional links for the switcher.
+   *
+   * @return array
+   *   An array of additional links keyed with the route name.
+   *
+   * @throws \Drupal\Component\Plugin\Exception\PluginNotFoundException
+   */
+  protected function getAdditionalLinks() {
+    return [
+      'entity.team.collection' => $this->t('Manage @teams', [
+        '@teams' => $this->entityTypeManager->getDefinition('team')
+          ->getPluralLabel(),
+      ]),
+    ];
+  }
+
+  /**
+   * Helper to find the current context from the route.
+   *
+   * @return mixed
+   *   The current context from the route or the route name or null.
+   *
+   * @throws \Drupal\Component\Plugin\Exception\PluginNotFoundException
+   */
+  protected function getCurrentContext() {
+    // Get the current context from the team or the user in the route.
+    if (($current_context = $this->routeMatch->getParameter('team'))
+      || ($current_context = $this->routeMatch->getParameter('user'))
+    ) {
+      return $current_context;
+    }
+
+    // Get the current context from additional links.
+    $additional_links = $this->getAdditionalLinks();
```

Comment:
So I think this would be a UX issue. The user sees the switcher block, clicks on it to navigate to the `/teams` page which then has no switcher. Wouldn't this break the workflow?

Reply:
Outdated. We are now showing the block on routes with corresponding route only.